src/contract/assignments.rs

```diff
@@ -404,7 +629,7 @@ impl AutoConceal for Assignments {
 }
 
 impl CommitEncodeWithStrategy for Assignments {
-    type Strategy = commit_strategy::UsingStrict;
+    type Strategy = commit_strategy::UsingConceal;
 }
 
```

Comment:
```suggestion
```

These lines should be deleted since they are not in effect. The implementation of the `CommitEncode` trait for the `Assignments` type is not needed. The new `Assignments::consensus_commits()` associated function is being used to do the required processing instead.

Reply:
There is an important difference between commitment encoding and consensus commitment. The first produces a vector of serialized data: the actual message we are comitting to. The second is produced from the first and is always a well-defined structure (like hash) that can be verified against the original message. You should not have consensus commit without commitment encoding.

Reply:
So:
1. We conceal the data (of they must or should be concealed according to the spec)
2. We commit-encode (serialize) the resulting concealed data as a byte string
3. We produce a commitment to those data (MerkleNode or other tagged hash form)
4. We merklize the tree of such MerkleNode commitments
5. We serialize the root of the Merkle tree
6. We take that serialized root as a consensus commit (since its already a hash and a commitment)

Thus we use concealment, encoding, commitment , merklization, encoding and shallow commitment for all our collections.

Reply:
Ok, you are right, I double-check the code and we cant produce any single commitment from the assignments; it is created only on the level above when we have assignments if all types. I think Assignments name here is very confusing, since this function covers only assignments of certain type.